filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/stream/config/AkkaRunnableGraph.java

```diff
@@ -354,4 +343,65 @@ AuditServiceSinkFactory auditServiceSinkFactory(
                         .map(committableTokenOffset -> (Committable) committableTokenOffset.t1())
                         .to(committerSink));
     }
+
+    @Bean
+    RunnableGraph<Control> tokenErrorMessageRunnableGraph(final Source<CommittableMessage<String, AuditErrorMessage>, Control> tokenErrorMessageSource,
+                                                               final Sink<Committable, CompletionStage<Done>> committerSink,
+                                                               final ErrorMessageEventService errorMessageEventService) {
+        return tokenErrorMessageSource
+                // Extract committable, token and message
+                .map(committableMessage -> Tuple3.create(
+                        committableMessage.committableOffset(),
+                        new String(committableMessage.record().headers().lastHeader(Token.HEADER).value(), Charset.defaultCharset()),
+                        committableMessage.record().value()
+                ))
+
+                // Filter out anything from this service
+                .filter(auditErrorMessage -> !auditErrorMessage.t3().getServiceName().equals(SERVICE_NAME))
```

Comment:
I feel this is quite an important point as it prevents that feedback-loop-like thing where the service returns errors about itself to the client.
I'm not sure how I'd describe and document it better, but if you can think of a good way to do it that would be good

Reply:
added, but probably needs a bit of refining